### ISSUE TITLE
BUG: 100% usage of available CPU upon "import numpy"

### ISSUE THREAD

**Issue by AyrtonRicardo** [numpy/numpy#25360]
### Describe the issue:

When doing simple `import numpy as np` the cpu goes to 100% usage in 7 out of 8 cores available.

### Reproduce the code example:

```python
import numpy
```

### Error message:

```shell
Here a video showing it since there is no error code, just CPU 100%: https://www.youtube.com/watch?v=cba5ntdMRkM
```

### Python and NumPy Versions:

> import sys, numpy; print(numpy.__version__); print(sys.version)
1.26.0
3.11.7 (main, Dec  8 2023, 18:56:58) [GCC 11.4.0]

### Runtime Environment:

> import numpy; print(numpy.show_runtime())
[{'numpy_version': '1.26.0',
  'python': '3.11.7 (main, Dec  8 2023, 18:56:58) [GCC 11.4.0]',
  'uname': uname_result(system='Linux', node='localhost', release='3.10.61', version='#1 SMP PREEMPT Tue Dec 5 22:24:50 UTC 2023', machine='aarch64')},
 {'simd_extensions': {'baseline': ['NEON', 'NEON_FP16', 'NEON_VFPV4', 'ASIMD'],
                      'found': [],
                      'not_found': ['ASIMDHP', 'ASIMDFHM']}},
 {'architecture': 'armv8',
  'filepath': '/home/android/hass/lib/python3.11/site-packages/numpy.libs/libopenblas64_p-r0-17488984.3.23.dev.so',
  'internal_api': 'openblas',
  'num_threads': 8,
  'prefix': 'libopenblas',
  'threading_layer': 'pthreads',
  'user_api': 'blas',
  'version': '0.3.23.dev'}]
None

### Context for the issue:

100% cpu usage is not health and I don't want to limit threads to a lower number because it will be too much of a cap in the application I'm using.

Numpy 1.26.0 is requirement of the application I'm running(Home Assistant), so I can't change it, but I did reproduce same issue on 1.26.2

**Comment by AyrtonRicardo** [comment]
Solution found!!!! It's related to the blas lib used.

> TL;DR: When installing `homeassistant` via pip, it requires `numpy==1.26.0`, which makes pip install `numpy` using default configuration as well as default `blas = openblas`. In openblas there is a bug of cpu using 100% of the available cpu. Fix is to remove numpy and install it using other blas lib (atlas worked to me).


Now for the detailed stuff:
To fix my problem after a ton of debug, I've found that the cause was numpy(simple import causes it to 100% cpu), so I found that the issue is caused by the blas lib that numpy was using, let's reinstall numpy using a better blas lib(in my case):

- Go inside your virtual env
- Remove numpy `pip uninstall numpy`
- Now we need to install our blas libraries, in my case ubuntu 22.04 arm64: `sudo apt install "*openblas*" pkg-config`. We also install `pkg-config` to be used instead of CMake. This will install around 4 openblas libraries, so we can choose the ones we want, even test them.
- Now, let's see what alternatives we have: `ls -lah /etc/alternatives/libblas*` and configure ATLAS to all of the available ones. In my case:
  - `sudo update-alternatives --config libblas.so-aarch64-linux-gnu`
  - `sudo update-alternatives --config libblas.so.3-aarch64-linux-gnu`
  - `sudo update-alternatives --config libblas64.a-aarch64-linux-gnu`
  - `sudo update-alternatives --config libblas64.so-aarch64-linux-gnu`
  - `sudo update-alternatives --config libblas64.so.3-aarch64-linux-gnu`
- You might want to remove libopenblas from your system `ls -lah /etc/alternatives/libopenblas*`:
  -  `sudo update-alternatives --config libopenblas.a-aarch64-linux-gnu`
  - `sudo update-alternatives --config libopenblas64.so`
  - `sudo update-alternatives --config libopenblas64.so.0-aarch64-linux-gnu`
  - `sudo update-alternatives --config libopenblas64.so-aarch64-linux-gnu`
  - `sudo update-alternatives --config libopenblas.so.0-aarch64-linux-gnu`
- Let's now install the correct version of numpy for Home Assistant, currently `1.26.0`: `pip install numpy==1.26.0 --no-binary numpy`. This command is gonna tell numpy during installation to use the global BLAS lib, which we are gonna use ATLAS or any other except openlibblas.
  - You might also need to install other libraries like `pip install ninja`
 


After this step by stepy I hope that you're able to reduce the cpu usage like I did: 
![image](https://github.com/numpy/numpy/assets/3937889/7313a2b5-c36e-45bc-8a52-567ee68779a1)



Also mentioned here: https://github.com/home-assistant/core/issues/105397#issuecomment-1848928627

**Comment by AyrtonRicardo** [comment]
Also opened a issue to openblas: https://github.com/OpenMathLib/OpenBLAS/issues/4365
